sdk/src/main/java/com/google/cloud/dataflow/sdk/util/Reshuffle.java

```diff
@@ -60,11 +54,14 @@ private Reshuffle() {
     // If the input has already had its windows merged, then the GBK that performed the merge
     // will have set originalStrategy.getWindowFn() to InvalidWindows, causing the GBK contained
     // here to fail. Instead, we install a valid WindowFn that leaves all windows unchanged.
-    Window.Bound<KV<K, V>> rewindow = Window
-        .<KV<K, V>>into(new PassThroughWindowFn<>(originalStrategy.getWindowFn()))
-        .triggering(new ReshuffleTrigger<>())
-        .discardingFiredPanes()
-        .withAllowedLateness(Duration.millis(BoundedWindow.TIMESTAMP_MAX_VALUE.getMillis()));
+    Window.Bound<KV<K, V>> rewindow =
+        Window.<KV<K, V>>into(
+                new IdentityWindowFn<>(
+                    originalStrategy.getWindowFn().windowCoder(),
+                    originalStrategy.getWindowFn().assignsToSingleWindow()))
+            .triggering(new ReshuffleTrigger<>())
+            .discardingFiredPanes()
+            .withAllowedLateness(Duration.millis(BoundedWindow.TIMESTAMP_MAX_VALUE.getMillis()));
```

Comment:
The coders here are unchanged in the underlying graph, so it should be update compatible.